### DISCUSSION TITLE
Developer resource: Widget for latest output data

### DISCUSSION THREAD

**Discussion by pyZerrenner** [pymeasure/pymeasure#1119] [Ideas]
While writing a measurement program with PyMeasure, I developed a couple of features that I would like to share with the community. They are not very user friendly, so I provide the scripts "as-is" and you can use them at you own risk. I encourage you to first understand the code before implementing it in your program.

My first feature is a new tab in the managed window, that simply displays the latest value that has been recorded by the currently running measurement. I find it helpful to have a live overview of all relevant quantities at once. Especially data that is relevant for running my experiment and for safety considerations (in my case pressure, temperature and water flow rate). You can specify which outputs from the procedure's `DATA_COLUMNS` to display and how many decimal digits to use.

The below zip-file consists of two scripts
- _latest_output_widget.py_: Contains the `LatestOutputWidget` that is added to the `widget_list` of a `ManagedWindow`. 
- _MWE_LatestOutput.py_: A minimal working example , that shows the functionality of the widget.

[latest_output_widget.zip](https://github.com/user-attachments/files/15749806/latest_output_widget.zip)

Preview screenshot of the MWE:
![MWE_Screenshot_240609](https://github.com/pymeasure/pymeasure/assets/47388301/86f3305b-6f08-4730-aa5a-8ac1af822ef1)

**Comment by CasperSchippers** [comment]
Nice tool, thanks for sharing your work!! Nice to see new tab widgets!
